### DISCUSSION TITLE
Question : how can users support GO?

### DISCUSSION THREAD

**Discussion by Psi58** [GrandOrgue/grandorgue#1746] [General]
It's a remarkable project and I, presumably like many others, congratulate the developers, and want it to prosper and continue.
I'm not a professional organist, not a coder, not even someone who gets GitHub well enough to offer updates to help files etc. - but I'd like to support it in some way.

I would be happy to contribute a little to the development (and developers) to keep servers, tools etc up to date - but (a) open source is a key GO strength (not to be given up lightly), and (b) I'm a pensioner with limited resources! 

Could there be something like an arms-length Foundation which users could be invited/encouraged to contribute to, eg? (TBH I've no idea how such a thing would be set up).

If so, perhaps that might also be a vehicle to negotiate a one-off payment for the release of unencrypted versions of samplesets currently unavailable to GO users - thus making the project even more attractive to new users.

Just my thoughts - interested to see how others feel.

**Comment by obw-go** [comment]
I'm using the GrandOrgue software for some years now and I'm very happy how well it works with a lot of different organ samples.
Also not being able to contribute to the technical development, I would like to contribute financially, because I suppose keeping up all the available stuff in the air will not be free from costs. 
I looked around on internet, trying to find information about providing a donation, but I can't find any.
Could anyone give me a hint where I coud find this?
Thanks!

**Reply by rousseldenis** [reply]
Hello @obw-go.

Thanks for your message. In fact, GrandOrgue is essentially supported by free time contributors (as many open source projects).

We switched to github three years ago as it offers  quite good infrastructure to maintain open source projects like this one for free 😃 

For now, it is based on individuals and is not formed into an association nor foundation as we don't need it yet.

If you want to contribute, you can do it through several ways:

- Diffuse the project around you (locally, social media,...)
- Report issues here (bugs, feature requests)
- Join discussions
- If you are tech oriented, propose changes through Pull Requests

All those kind of things are, in fact, contributions. If you have further questions, don't hesitate! I hope this could have answered the first ones.

**Reply by obw-go** [reply]
Hi Denis, 

Thanks for your quick response, which gives me a good explanation of the open source character of the GrandOrgue project.

It feels a bit strange to transfer money to people who make beautiful samples for GrandOrgue and nothing to the builders of GrandOrgue itself...

Anyway, the diffusion of GrandOrgue is already going well - I have already put several organ enthusiasts on the trail of GrandOrgue.

As for other ways to contribute, I unfortunately don't have the time for the time being - hopefully I can tackle that in the future.

wkr, Otto

**Reply by rousseldenis** [reply]
> It feels a bit strange to transfer money to people who make beautiful samples for GrandOrgue and nothing to the builders of GrandOrgue itself...

I think good sample sets take a huge amount of time until release. The limit is fixed by the creator between free release or not. E.g.: Piotr Grabowski released his first samples freely. You can have also great free samplesets of @larspalo : https://familjenpalo.se/vpo/about-my-samplesets/

But the model can evolve. I'm personally in another open source community that needs money contributions as there is a lot of contributors/users that need: a bigger technical infrastructure, events organization,...

**Reply by larspalo** [reply]
If you want my opinion and/or advice a as both a developer of GO and mostly as a free sample set producer and programmer of tools to help with that task... Pay the sample set producer what they want to get the sample set that you want to use! Life is short, and mostly, money is very cheap compared to what you can get in terms of the experience...

The cost of producing a sample set is, at least in my experience, way higher than what any of the sample set producers I've looked at actually are charging for their products... If you're not happy with what they are charging, or what sample sets you can get for free, you're very much welcome to try to produce the equivalent product "for free"!!!

**Reply by obw-go** [reply]
Hi Lars, 

Thanks for your response as well! 

I was just looking on your site https://familjenpalo.se/vpo/  checking out what you tell about all the necessary steps to create a sample set.  You make it very clear that this process requires a breathtaking amount of time and careful attention to details.
I consider myself lucky that I don't have the time for that at the moment... ;)

I have already paid for sample sets several times and your explanation motivates me to continue to do so more often in the future.

wkr, Otto

**Comment by larspalo** [comment]
Jokes, aside! If anyone wants to contribute to the GrandOrgue project, the most important contribution you can give is using it! Report your findings, and if you have the desire, ability and/or capability of doing so, help the project by devoting you time and energy towards it in any way you can!

**Comment by oleg68** [comment]
@obw-go Thank you for your intention to help us! Unfortunately we do not accept donations. But any other help Lars mentioned is welcome.

**Comment by obw-go** [comment]
@lars, @roussell @oleg68 
What I can do is some editting on Wikipedia pages related to GrandOrgue. To my surprise there was since 2011 wrong information about GrandOrgue on the Dutch wiki page about Hauptwerk, stating that the GrandOrgue software was not maintained since 2006. See the attacht picture, which shows the change I made on this page.  
![Wikipedie_Hauptwerk_wrong_info_about_GrandOrgue](https://github.com/user-attachments/assets/ea102e91-2923-4138-a183-9cb897afd56e)

**Comment by obw-go** [comment]
There was no Dutch wiki page about GrandOrgue, so I did add this page as well. See: https://nl.wikipedia.org/wiki/GrandOrgue

**Comment by obw-go** [comment]
You can check the contents of the page by having it translated in your langauge. Please let me know it something shoud be changed in its content...

**Reply by larspalo** [reply]
Thanks for providing a wikipedia page in Dutch for GrandOrgue! From the english translation of it, the only thing I'm slightly critical of is that in one sentence where it seems that GO has a wind model already in place:

```
Onder andere door gebruik van algoritmes die de winddruk van het pijporgel nabootsen, is het effect, afhankelijk van het register, het aantal toetsaanslagen en de snelheid van de gespeelde noten van invloed op de levendigheid van het geluid.
```

This is not exactly true, at least in the way as the english translation of it makes it sound like, as we have yet to develop/implement such a feature. But there are many other features in GO that improves the realism and/or playing experience: possible use of multiple attack/release samples, usage of recorded tremulant samples if available, possibility to use multi channel audio distribution of samples and usage of impulse response files to provide ambience, to just name a few.

An other possible help you perhaps could provide is checking the nl.po translation of GO that already exist, and if necessary, learn to provide an updated version of that translation file. For instance to provide the currently non translated strings or fix previous errors in the translation.

**Reply by obw-go** [reply]
Hi Lars, 
I replaced the premature sentence by the info you provided. 

What do you mean by **nl.po**? Is that the name of the Dutch language file in which I can supplement and correct Dutch strings if necessary? If so, where can I find this file? Nice to do... Years ago I did the same for the wellknown adblocker AdBlock.
wkr, Otto

**Comment by obw-go** [comment]
Ah, I found the file already: https://github.com/GrandOrgue/grandorgue/blob/master/po/nl.po
Can I directly make changes in this file or should I do it another way?
And is Gerrit Veldman no longer active as translator?

**Reply by larspalo** [reply]
The https://github.com/GrandOrgue/grandorgue/blob/master/README.translate contains information about how to update the translation file from the most recent source, but it does involve first building the app, which is most easily done on Linux (in my opinion). I've personally used poedit for updating the .po files after building.

The easiest way could perhaps be to fork of the GrandOrgue repository, clone the fork to your local computer, build it, then update the translation file according to the information in the README.translate file, do the updating of the nl.po strings in any editor of your choice, push the updated file to your fork and then create a pull request to the official GrandOrgue repository.

I can understand that it sounds like a lot of steps that might be difficult to comprehend, but it's really not at all that difficult to do.

**Reply by obw-go** [reply]
Thx for you quick reply - I'll first study the README.translate...

**Comment by obw-go** [comment]
Hi Lars,

Your info and README.translate invite me to make a "buillding of the app on Linux". Since I'm not an opensource programming developer and use a Windows PC, it seems that I cannot take this detour and create a version of nl.po that contains the current strings from the recent program version added. Of course the most recent strings from the more recent versions of the program are missing in e.g. the present nl.po in the master directory...

Would it be possible for one of the developers to add an extra button in the header of the .po files, which creates a download in the selected language with all current strings of the most recent program with their (partly missing) translations?
I could then modify this file in an editor on my PC and put the file in a pull request...

wkr, Otto

**Reply by larspalo** [reply]
@obw-go Almost anything "could" be done...

However, meanwhile you can try using this zipped nl.po file that have all the new strings included and see if you could update the text in it... I do however recommend using something like poedit (which seems to have builds for windows) for that. 
[nl.zip](https://github.com/user-attachments/files/16236264/nl.zip)

**Reply by obw-go** [reply]
Hi Lars, 
After downloading and installing poedit, I understand your recommendation: the program corresponds to the specific .po files and offers several facilities for providing the translations (e.g. indicating what is wrong).
I'm starting to work on it.
wkr, Otto

**Comment by obw-go** [comment]
@larspalo 
Hi Lars,
Looking around in the 'Pull request' context I did not find a way to drop the (supplemented) Dutch translation nl.po in there.
So I have to use a workaround.

Attached you will find the nl.po with the (supplemented) Dutch translations as performed with poedit.
Oops -  "We don’t support that file type" is the message I get in this 'Discussions context'.

workaround 2:  
I copied nl.po to my cloud server, so it can be downloaden through the next link:
https://ottobw.stackstorage.com/s/IwSJZ28R5zQE2a0r
wkr, Otto

**Reply by larspalo** [reply]
@obw-go The way to do this is to first fork the GrandOrgue repository to your Github account.

![CreateFork](https://github.com/user-attachments/assets/8cf7caad-c3c2-4c3a-814d-d49cdd9cd459)

Then you can easily upload your nl.po file to the /po directory of your own fork in the Github UI. (Hint, click "Add file" in the po/ directory of your fork).

Next you can create a pull request to the official GrandOrgue repository from your fork for the translation file to be included.

In that way you'll also automatically be recognized as a contributor for that specific translation (changes/updates), whereas if anyone else would upload the file you've translated you wouldn't be. It might not make any difference to you personally, but I somehow think it's great to be able to, from the commit history, see who did a certain contribution.

**Comment by obw-go** [comment]
@larspalo 
Thanks for your hints. After taking the wrong fork within my fork, I seem in another try to have taken the right fork for committing the pull request...
wkr, Ottto

**Comment by obw-go** [comment]
Ah, I see now that there is also a topic ''Dutch Translation'', so it's more on topic to continue this discussion regarding nl.po in that topic.

**Comment by obw-go** [comment]
@larspalo 
What is on topic here is my next question: would it be a good idea to provide on your page 
https://familjenpalo.se/vpo/download/ 
also an option to transfer a donation for the use of the sampleset(s) you made?
like e.g. Piotr did on his page: 
https://piotrgrabowski.pl/donate/
